### ISSUE TITLE
Blank line inserted above all other content when switching prompts

### ISSUE THREAD

**Issue by rschristian** [brigand/glint#11]
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behaviour?**
Currently a blank line is inserted above prompt content when switching between prompts.

**If the current behaviour is a bug, please provide the steps to reproduce.**
* Run `glint c` in a repository with unstaged changes
* After you select a file to be staged and press the `enter` key there will be a blank line above the type prompt content
* Press the `escape` key to go back to the files prompt and there will be a second blank line
* This repeats forever

![Peek 2020-10-01 01-09](https://user-images.githubusercontent.com/33403762/94774140-dd65cc00-0382-11eb-94bd-f4a89bbb79d4.gif)

**What is the expected behaviour?**
No new blank lines

**Please mention other relevant information.**
I'll try to track this down myself and make a PR if I can find & fix this. This issue is mainly for if I cannot, so at least the bug is noted.

**Comment by rschristian** [comment]
So I've seem to track this down to the cursor placement. 

When a user starts Glint, the cursor will be on the whatever you call the starting line. However, when a full frame is being rendered, the cursor is only moved up the height of the content, so it will be one line below (seriously, if there's a name for that, let me know).

```
$ glint c  <-- cursor
<content> <-- cursor y = 1
```

So from the bottom of the content (2) moving up 1 places you below the starting line. Then you clear everything below that point, start a new line, and print the content all the same.

```
$ glint c
                   <-- cursor moved here
<content> 
```

This is going to take some playing around, but what I think I want to do is move the cursor up `y + 1` and to the right say `1000`. This puts it at the very right bound of the `$ glint c` line. We can then clear everything below, move back left to a reasonable starting position and then run from there.

**Comment by rschristian** [comment]
Might be a bit fiddle without adding in more rendering logic. For example, while the gif above is no longer reproduceable, if I were to press the `enter` key on the file prompt with nothing staged the frame will be rerendered one position higher.

![Peek 2020-10-09 02-22](https://user-images.githubusercontent.com/33403762/95554709-40381280-09d6-11eb-8c54-577c89187bce.gif)

It also appears to massacre terminal formatting... so there's that. 

![temp](https://user-images.githubusercontent.com/33403762/95554796-66f64900-09d6-11eb-9b15-0220375ee012.png)

Bugs to fix but progress being made anyhow.

**Comment by rschristian** [comment]
Fixed with #17
